staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -136,10 +138,25 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 
 	envVarFiles := os.Getenv(RecommendedConfigPathEnvVar)
 	if len(envVarFiles) != 0 {
-		chain = append(chain, filepath.SplitList(envVarFiles)...)
-
+		files := filepath.SplitList(envVarFiles)
+		for ix := range files {
+			moreFiles, err := loadFileOrDir(files[ix])
+			if err != nil {
+				glog.V(3).Infof("Failed to load config %s (%v)", files[ix], err)
```

Comment:
(I had to leave one error message in there, a call to `os.Stat` which should really never return an error unless something super bad happens. It is necessary b/c comments above want the directory to only be used if the file doesn't exist, and there's no way to now except in that function)